### ISSUE TITLE
it is possible to get collection async?

### ISSUE THREAD

**Issue by YaleZheng** [ecrmnn/collect.js#132]
i want something like 
```javascript
const collect = require('collect.js')

collect(['a', 'b', 'c'])
    .toArray(array => /** do something */)
```

it is possible to achieve with current version?

**Comment by YaleZheng** [comment]
after reading source code.i think i found i could achieve my purpose by
```javascript
const collect = require('collect.js')
collect(['a', 'b', 'c'])
    .chunk(Number.MAX_VALUE)
    .each(array => /** do something */)
```
but it look really ugly! is there any better way?

**Comment by ecrmnn** [comment]
Do you have a real world example?
I don't see what you're trying to achieve.

**Comment by YaleZheng** [comment]
i just want to do everything without leaving the chain,but i could not found some straight way to achieve
that.

```javascript
//sync
const arr = collect(['a', 'b', 'c'])
                        /** do some transform */
                        .toArray()
/** do something with array */

//async
collect(['a', 'b', 'c'])
    .chunk(Number.MAX_VALUE)
    .each(array => /** do something */)
```
in my opinion,the async way may be more clean,although the sync way work too

**Comment by ecrmnn** [comment]
Have you looked at ``map()`` or ``transform()``?

```js
collect([1, 2, 3, 4])
  .transform(i => i * 2)
  .dump() //=> [2, 4, 6, 8]
  .transform(i => i * 2)
  .all();

//=> [4, 8, 12, 16]
```

I noticed that ``transform()`` doesn't return the modified collection in 3.0.50, but it's fixed in 3.0.51. So please upgrade or use ``map()``.

**Comment by YaleZheng** [comment]
Thks for your help,I have soluted this problem~XD
By the way,I found an performance issues,I will post it later
